### DISCUSSION TITLE
Serverside Data Table for Big Data Handling

### DISCUSSION THREAD

**Discussion by Zen225** [primefaces/community#891] [PrimeReact] [answered]
Hello.

I want to handle some big Data from my Application where base on next JS, but if i use the Original Datatable from PrimReact it was client side method i guess

maybe the clientside method cant handle big data, any tutorial or recommendation for handling big data with datatable primereact

Many Thank's
regards

**Comment by melloware** [comment, marked as answer]
I am so glad I wrote this as I answered this question a ton of times.  I have an example showing using REST Services to use the `lazy` feature of the datatable so it Sorts, Filters, Paginates etc Just in time.  Running queries to show the bare minimum to represent the table.

Full working example with Code and in memory DB: https://github.com/melloware/quarkus-primereact

In the example I load 500 DB records but you could load 50,000 and it would be the same performance.

**Comment by mertsincan** [comment]
And https://dev.to/mertsincan/handling-big-data-on-datatable-with-checkbox-selection-multiselect-4hgc

Best Regards,
